sys/net/gnrc/network_layer/ipv6/netif/gnrc_ipv6_netif.c

```diff
@@ -513,10 +512,14 @@ kernel_pid_t gnrc_ipv6_netif_find_by_prefix(ipv6_addr_t **out, const ipv6_addr_t
         mutex_unlock(&(ipv6_ifs[i].mutex));
     }
 
+    if (out != NULL) {
+        *out = found;
```

Comment:
Since the return value already indicates that a result wasn't found, the old method should suffice, since you never should use results of a failed function anyway. The value could be anything before the execution, too. We should however document the behavior, that the function might change `out` even in an unsuccessful case.